File: lab1.java
/* 
   This program is a decrypter that takes in an argument, 
   an existing plain-text phrase and returns encrypted messages.
*/
import java.util.Scanner;


public class lab1 {

    /*
    * This method takes in a msg String and prints out the decoded msg.
    */
   
   public static void decode(String msg) {
       char[] encodedArray = msg.toCharArray(); //puts msg from parameter to a char array 
       char[] decodeArray = new char[encodedArray.length]; //made a new char array to input the decoded msg.
       int intChar; //used to get integer value of character and store it
       int a = 97; //ascii of 'a' value 

       for (int i = 0; i < 26; i++) // iterating through the alphabet 
       {
           for (int k = 0; k < encodedArray.length; k++) // iterating through the charArray
           {

               if (encodedArray[k] != ' ') {   //if the element in index k is not a space 
                    intChar = encodedArray[k];  //store the value of the element 
                    intChar = (intChar - a + i) % 26 + a;  //use this formula to figure out the correct decoded msg from the element at index k for the alphabet at index i
                    decodeArray[k] = (char)intChar; //finally add the char of the integer value to the new decoded array

               } else {
                   decodeArray[k] = ' '; //if not an element in index k, it is a space 
               }
           }
           System.out.println(new String(decodeArray) + "\n"); //prints out array in form of a string instead of iterating through each value of array to print it out 
       }

   }

   public static void main(String[] args) {
       Scanner scan = new Scanner(System.in); // scanner for input from user
       int input; // input from user for Question we want to do
       String toDecode; //msg to decode 

       System.out.println("Which Q. did you want to do? (1-4)");
       input = scan.nextInt(); //takes users input

       switch (input) { //switch statement for easy change of inputs 
           case 1:
               toDecode = "fqjcb rwjwj vnjax bnkhj whxcq nawjv nfxdu mbvnu ujbbf nnc";
               decode(toDecode);
               break;

           case 2:
               toDecode = "oczmz vmzor jocdi bnojv dhvod igdaz admno ojbzo rcvot jprvi oviyv aozmo cvooj ziejt dojig toczr dnzno jahvi fdiyv xcdzq zoczn zxjiy";
               decode(toDecode);
               break;

           case 3:
               toDecode = "ejitp spawa qleji taiul rtwll rflrl laoat wsqqj atgac kthls iraoa twlpl qjatw jufrh lhuts qataq itats aittk stqfj cae";
               decode(toDecode);
               break;

           case 4:
               toDecode = "iyhqz ewqin azqej shayz niqbe aheum hnmnj jaqii yuexq ayqkn jbeuq iihed yzhni ifnun sayiz yudhe sqshu qesqa iluym qkque aqaqm oejjs hqzyu jdzqa diesh niznj jayzy uiqhq vayzq shsnj jejjz nshna hnmyt isnae sqfun dqzew qiead zevqi zhnjq shqze udqai jrmtq uishq ifnun siiqa suoij qqfni syyle iszhn bhmei squih nimnx hsead shqmr udquq uaqeu iisqe jshnj oihyy snaxs hqihe lsilu ymhni tyz";
               decode(toDecode);
               break;

           default:
               System.out.println("Wrong input please try again");
               break;

       }

       scan.close(); //close scanner to prevent mem leak

   }
}

File: lab1.py
enc_msg = [
    "fqjcb rwjwj vnjax bnkhj whxcq nawjv nfxdu mbvnu ujbbf nnc",
    "oczmz vmzor jocdi bnojv dhvod igdaz admno ojbzo rcvot jprvi "
    "oviyv aozmo cvooj ziejt dojig toczr dnzno jahvi fdiyv xcdzq zoczn zxjiy",
    "ejitp spawa qleji taiul rtwll rflrl laoat wsqqj atgac kthls "
    "iraoa twlpl qjatw jufrh lhuts qataq itats aittk stqfj cae",
    "iyhqz ewqin azqej shayz niqbe aheum hnmnj jaqii yuexq ayqkn "
    "jbeuq iihed yzhni ifnun sayiz yudhe sqshu qesqa iluym qkque "
    "aqaqm oejjs hqzyu jdzqa diesh niznj jayzy uiqhq vayzq shsnj "
    "jejjz nshna hnmyt isnae sqfun dqzew qiead zevqi zhnjq shqze "
    "udqai jrmtq uishq ifnun siiqa suoij qqfni syyle iszhn bhmei squih "
    "nimnx hsead shqmr udquq uaqeu iisqe jshnj oihyy snaxs hqihe lsilu ymhni tyz"
    ]


dec_msg = ""

#print(enc_msg[0], "\n")
#print(enc_msg[1], "\n")
#print(enc_msg[2], "\n")
#print(enc_msg[3], "\n")

#letter frequency
lf = {}

for keys in enc_msg[1]:
    if keys != ' ':
        lf[keys] = lf.get(keys, 0) + 1

print(lf)

max_letter = max(lf, key=lambda key: lf[key])

#most used english letters
muel = ['e', 't', 'a', 'o', 'i', 'n', 's', 'h', 'r', ]
for i in muel:
    dec_msg = ''
    #offset = (ord(max_letter) + 26 - ord(muel[8]))%26
    offset = (ord(max_letter) + 26 - ord(i))%26
    #print(offset)

    for letter in enc_msg[1]:
        if letter != ' ':
            dec = ((ord(letter) - ord('a') + 26 - offset)% 26) + ord('a')
            #print(dec, chr(dec)
        else:
            dec = ord(' ')
        dec_msg += chr(dec)

    print(i, dec_msg, "\n")
